robocop/checkers/misc.py

```diff
@@ -1566,3 +1644,47 @@ def check_for_complex_condition(self, condition_token, node_name, left_side, var
                 col=position,
                 end_col=position + len(variable) + len(right_side),
             )
+
+
+class TestAndKeywordOrderChecker(VisitorChecker):
+    reports = ("test-case-section-out-of-order", "keyword-section-out-of-order")
+
+    def __init__(self):
+        self.rules_by_node_type = {}
+        self.expected_order = {}
+        super().__init__()
+
+    def visit_File(self, node):  # noqa
+        self.rules_by_node_type = {Keyword: "keyword-section-out-of-order", TestCase: "test-case-section-out-of-order"}
+        self.expected_order = {
+            Keyword: self.param("keyword-section-out-of-order", "sections_order"),
+            TestCase: self.param("test-case-section-out-of-order", "sections_order"),
+        }
+        self.generic_visit(node)
+
+    def check_order(self, node):
+        if type(node) not in self.rules_by_node_type:
```

Comment:
This code may not be needed. If you're only calling this method in visit_Keyword and visit_TestCase, the node type will be always either Keyword and TestCase.